src/south/ProtocolHandler.class.js

```diff
@@ -101,9 +110,26 @@ class ProtocolHandler {
    * @return {void}
    */
   addValues(values) {
```

Comment:
As Szabi mentionned, we can have data loss. Maybe we can pass the lastCompletedAt and the dataSourceId as parameters, store them in an array and update the database at the end of the flush to fix that